src/interpolate_years.py

```diff
@@ -0,0 +1,96 @@
+import logging
+import hydra
+import pandas as pd
+import numpy as np
+
+logger = logging.getLogger(__name__)
+
+def weighted_mean(values, weights):
+    return np.average(values, weights=weights)
+
+def weighted_sum(values, weights):
+    return np.sum(values * weights)
+
+@hydra.main(config_path="../conf", config_name="config", version_base=None)
+def main(cfg):
+    # read in dec00 and acs09 file
+    min_yr = cfg.interp.min_interp_year
+    max_yr = cfg.interp.max_interp_year
+
+    sp_res = cfg.interp.sp_res
+    var_dict = cfg.variables.valid_years[sp_res]
+
+    # construct survey dict
+    # Start with dec years
+    survey_dict = {yr: "dec" for yr in var_dict["dec"]}
+    # Add acs5 years only if not already in the dict
+    for yr in var_dict["acs5"]:
+        if yr not in survey_dict:
+            survey_dict[yr] = "acs5"
+
+    df_start = pd.read_parquet(f"{cfg.interp.input_dir}/{survey_dict[min_yr]}_{min_yr}.parquet")
+    df_end = pd.read_parquet(f"{cfg.interp.input_dir}/{survey_dict[max_yr]}_{max_yr}.parquet")
+
+    year_lst = range(min_yr, max_yr+1)
+    year_gap = len(year_lst)
+    var_lst = list(cfg.variables.names.keys())
+
+    # read in crosswalk file
+    if cfg.interp.xwalk is not None:
+        zcta_xwalk = pd.read_parquet(cfg.interp.xwalk.xwalk_fname)
+        # merge with df_start
+        df_start = df_start.merge(zcta_xwalk, left_on="zcta", right_on="zcta_00", how="inner")
+        
+        # Keep only necessary columns
+        df_start = df_start[var_lst + ["weight", "zcta_10"]].copy()
+
+        rate_vars = cfg.interp.xwalk.rate_vars
+        count_vars = [v for v in var_lst if (v not in rate_vars)]
+
+        # Calculate weighted means and sums using groupby
+        # Use a list of aggregations for each variable
+        agg_dict = {
+            **{var: lambda x: weighted_mean(x, df_start.loc[x.index, "weight"]) for var in rate_vars},
```

Comment:
if weights add up to one, then the weighted mean and sum are no different. Weighted sum does not necessarily lead to absolute value (no decimal), if that is the intention @jckitch